src/core/component.js

```diff
@@ -259,10 +260,14 @@ module.exports.registerComponent = function (name, definition) {
   NewComponent = function (el, attr) {
     Component.call(this, el, attr);
   };
+
   NewComponent.prototype = Object.create(Component.prototype, proto);
   NewComponent.prototype.name = name;
   NewComponent.prototype.constructor = NewComponent;
   NewComponent.prototype.system = systems && systems.systems[name];
+  // Wraps method to meet needed gurantees
```

Comment:
`Wraps methods to meet needed guarantees.`

Can also explain what the "guarantees" are.